### ISSUE TITLE
Build error: environment can only contain strings

### ISSUE THREAD

**Issue by KellyThomas** [godotengine/godot#19739]
**Godot version:**
25275de with mono

**OS/device including version:**
Windows 7
Python 2.7.14
Mono 5.12

**Issue description:**
When I execute:
```
call scons.bat -j2 platform=windows module_mono_enabled=yes tools=yes  mono_glue=no
```
The build fails with this message:
```
MSBuild path: C:\Program Files\Mono\bin\msbuild.bat
scons: *** [bin\GodotSharpTools.dll] TypeError : environment can only contain st
rings
Traceback (most recent call last):
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Action.py", l
ine 1197, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "C:\Apps\godot\engine-repo\modules\mono\SCsub", line 206, in mono_build_s
olution
    subprocess.check_call(msbuild_args, env=msbuild_env)
  File "C:\Python27\lib\subprocess.py", line 181, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "C:\Python27\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
scons: building terminated because of errors.
```

This error does not occur when building 80f7eac with mono

**Comment by KellyThomas** [comment]
May be related to https://github.com/godotengine/godot/pull/19016

**Comment by RaXaR** [comment]
I'm also getting this error when I try to run
```
scons platform=windows tools=yes module_mono_enabled=yes mono_glue=no
```

I'm using x64:
* Windows 10
* VS 2017
* Mono 5.12.166
* Python 2.7.14

Note:
Mono is added to the PATH as "C:\Program Files\Mono\bin". 
I also ran the vc_vars batch file for Visual Studio x64:
"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars64.bat"

Build fails with the following output:
```
[Initial build] Linking Program        ==> bin\godot.windows.tools.64.mono.exe
   Creating library bin\godot.windows.tools.64.mono.lib and object bin\godot.windows.tools.64.mono.exp
[Initial build] mono_build_solution(["bin\GodotSharpTools.dll"], ["modules\mono\editor\GodotSharpTools\GodotSharpTools.sln"])
MSBuild path: C:\Program Files\Mono\bin\msbuild.bat
scons: *** [bin\GodotSharpTools.dll] TypeError : environment can only contain strings
Traceback (most recent call last):
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Action.py", line 1197, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "G:\Programming\godot\modules\mono\SCsub", line 206, in mono_build_solution
    subprocess.check_call(msbuild_args, env=msbuild_env)
  File "C:\Python27\lib\subprocess.py", line 181, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "C:\Python27\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
CC @neikeq

**Comment by RaXaR** [comment]
Thanks @KellyThomas , I manually added your changes in the above PR to my local repo and it worked just fine. Thanks for the effort.

**Comment by RaXaR** [comment]
Ugh, wait, I might have spoken too soon.
I added the fixes in the PR and it solved the previous error, but now there is a new one:
```
> scons platform=windows tools=yes module_mono_enabled=yes mono_glue=no
scons: Reading SConscript files ...
Configuring for Windows: target=debug, bits=default
Found MSVC compiler: amd64
Compiled program architecture will be a 64 bit executable (forcing bits=64).
YASM is necessary for WebM SIMD optimizations.
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[100%] mono_build_solution(["bin\GodotSharpTools.dll"], ["modules\mono\editor\GodotSharpTools\GodotSharpTools.sln"])
MSBuild path: C:\Program Files\Mono\bin\msbuild.bat
scons: *** [bin\GodotSharpTools.dll] TypeError : environment can only contain strings
Traceback (most recent call last):
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Action.py", line 1197, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "G:\Programming\godot\modules\mono\SCsub", line 212, in mono_build_solution
    subprocess.check_call(msbuild_args, env=msbuild_env)
  File "C:\Python27\lib\subprocess.py", line 181, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "C:\Python27\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
scons: building terminated because of errors.
```

**Comment by KellyThomas** [comment]
@RaXaR Hmm...   I think it might be that in your build environment it also hits a later call to `monoreg.find_msbuild_tools_path_reg()`.

I've added another clause to handle that, can I please ask you to test the updated PR.

**Comment by RaXaR** [comment]
@KellyThomas thanks for the feedback. I went back and cleaned my repo (Git->clean up). 
Then I changed my env setup to use 
```
MONO64_PREFIX=C:\Program Files\Mono 
```
as per the docs here [Compiling with Mono](http://docs.godotengine.org/en/3.0/development/compiling/compiling_with_mono.html)

This solved the issue for me. I have not tried the latest change to your PR though. I can still do so if you want?
